src/core/public/application/types.ts

```diff
@@ -199,7 +200,7 @@ export type AppUpdater = (app: AppBase) => Partial<AppUpdatableFields> | undefin
  * Extension of {@link AppBase | common app properties} with the mount function.
  * @public
  */
-export interface App extends AppBase {
+export interface App<HistoryLocationState = any> extends AppBase {
```

Comment:
default `any` seems brutal as a default for something as specialized as `HistoryLocationState` ?

Reply:
This is the same default that `History` uses itself, but I agree it's not great. How do you feel about making it `unknown` that way applications that don't use history state don't have to specify a type argument, but any application that does **must** provide a type argument (or face casting it manually everywhere).

Reply:
Sorry you are right. Got tricked by
```ts
History<HistoryLocationState = LocationState>
```

but

```ts
type LocationState = any
```

`unknown` might be better, don't have a strong opinion on that though.

Reply:
Yeah that caught me up when I first started on this as well. Switched to `unknown` for now.